packages/react-aria-components/src/Group.tsx

```diff
@@ -10,19 +10,60 @@
  * governing permissions and limitations under the License.
  */
 
-import {ContextValue, useContextProps} from './utils';
+import {ContextValue, StyleRenderProps, useContextProps, useRenderProps} from './utils';
+import {mergeProps, useFocusRing, useHover} from 'react-aria';
 import React, {createContext, ForwardedRef, forwardRef, HTMLAttributes} from 'react';
 
-export const GroupContext = createContext<ContextValue<HTMLAttributes<HTMLElement>, HTMLDivElement>>({});
+export interface GroupRenderProps {
+  /**
+   * Whether the group is currently hovered with a mouse.
+   * @selector [data-hovered]
+   */
+  isHovered: boolean,
+  /**
+   * Whether an element within the group is focused, either via a mouse or keyboard.
+   * @selector :focus-within
+   */
+  isFocusWithin: boolean,
+  /**
+   * Whether an element within the group is keyboard focused.
+   * @selector [data-focus-visible]
+   */
+  isFocusVisible: boolean
```

Comment:
Debated making this `isFocusVisibleWithin` but seemed like it was getting kinda long. Wondered about making the previous one `isFocused` for consistency even though it is within too, but we're using a native CSS selector for that so maybe confusing? 🤔